glitch/analysis/security.py

```diff
@@ -1,15 +1,1107 @@
+                        if a:
+                            errors.append(Error('sec_public_ip', a, file, repr(a)))
+                        
+            elif isinstance(element, Attribute) or isinstance(element, Variable):
+                for config in SecurityVisitor._PUBLIC_IP_CONFIGS:
+                    if (elem_name == config['attribute'] and au_type in config['au_type']
+                        and parent_name in config['parents'] and not element.has_variable 
+                        and elem_value.lower() not in config['values']
+                        and config['values'] != [""]):
+                        return [Error('sec_public_ip', element, file, repr(element))]
+            return errors
+
+    class TerraformAccessControl(TerraformSmellChecker):
+        def check(self, element, file: str, code, elem_name: str, elem_value: str = "", au_type = None, parent_name = ""):
+            errors = []
+            if isinstance(element, AtomicUnit):
+                if (element.type == "resource.aws_api_gateway_method"):
+                    http_method = self.check_required_attribute(element.attributes, [""], 'http_method')
+                    authorization =self.check_required_attribute(element.attributes, [""], 'authorization')
+                    if (http_method and authorization):
+                        if (http_method.value.lower() == 'get' and authorization.value.lower() == 'none'):
+                            api_key_required = self.check_required_attribute(element.attributes, [""], 'api_key_required')
+                            if api_key_required and f"{api_key_required.value}".lower() != 'true':
+                                errors.append(Error('sec_access_control', api_key_required, file, repr(api_key_required)))
+                            elif not api_key_required:
+                                errors.append(Error('sec_access_control', element, file, repr(element), 
+                                f"Suggestion: check for a required attribute with name 'api_key_required'."))
+                    elif (http_method and not authorization):
+                            errors.append(Error('sec_access_control', element, file, repr(element), 
+                                f"Suggestion: check for a required attribute with name 'authorization'."))
+                elif (element.type == "resource.github_repository"):
+                    visibility = self.check_required_attribute(element.attributes, [""], 'visibility')
+                    if visibility:
+                        if visibility.value.lower() not in ["private", "internal"]:
+                            errors.append(Error('sec_access_control', visibility, file, repr(visibility)))
+                    else:
+                        private = self.check_required_attribute(element.attributes, [""], 'private')
+                        if private:
```

Comment:
For these cases (and for the reverse), it is better to use `is not None` (or `is None`).
Since Python is dynamically typed doing these checks helps in avoiding weird cases.
E.g. imagine if instead of None, the function returns False for some reason. If you check `not foo()`, this will return True. However, if you check `foo() is None`, it will return False.

Reply:
There are many cases such as this in this file.

Reply:
Sure. Did not think about that, but I agree.